crypto/rand/drbg_lib.c

```diff
@@ -141,6 +147,152 @@ static int is_digest(int type)
     }
 }
 
+/*
+ * Initialize the OPENSSL_CTX global DRBGs on first use.
+ * Returns the allocated global data on success or NULL on failure.
+ */
+static void *drbg_ossl_ctx_new(OPENSSL_CTX *libctx)
+{
+    DRBG_GLOBAL *dgbl = OPENSSL_zalloc(sizeof(*dgbl));
+
+    if (dgbl == NULL)
+        return NULL;
+
+    if (!CRYPTO_THREAD_init_local(&dgbl->private_drbg, NULL))
+        goto err1;
```

Comment:
I would prefer seeing the CRYPTO_THREAD_cleanup_local() implementations checked for NULL, rather than doing this.  It looks like basic code :)

Reply:
I don't believe this is possible. `&dgbl->private_drbg` here will never be NULL even if it has not been initialised by `CRYPTO_THREAD_init_local`. The contents of `dgbl->private_drbg` might have a zero value because it has been "zalloc'd". But we do not know whether 0 is an invalid value or not for the local platform threading scheme - so it would not be correct for `CRYPTO_THREAD_cleanup_local` to check for 0.